### PR TITLE
Feature/index based key store

### PR DESCRIPTION
**Pull request by Dark0096**
*Feature*
- add the index based key store handler for performance

Example) If you save `e.e.e.k` metric using by index based key store handler, saved like below:
image

### CODE REVIEW

File: graphene-common/src/main/java/net/iponweb/disthene/reader/utils/DateTimeUtils.java [hunk 1]
```diff
@@ -50,4 +50,8 @@ private static long getUnitValue(String s) {
     public static boolean testTimeOffset(String s) {
         return timeOffsetPattern.matcher(s.replaceAll("^['\"]|['\"]$", "")).matches();
     }
+
+    public static long currentTimeSeconds() {
```
Comment (jerome89) [inline graphene-common/src/main/java/net/iponweb/disthene/reader/utils/DateTimeUtils.java]: How about naming currentTimestamp instead?

File: graphene-writer/src/main/kotlin/com/graphene/writer/store/key/SimpleKeyStoreHandler.kt [hunk 2]
```diff
@@ -31,133 +14,25 @@ import javax.annotation.PostConstruct
...
-
-  private fun addToBatch(metric: GrapheneMetric) {
-    request.add(metric)
-
-    if (batchSize <= request.size() || lastFlushTimeSeconds < currentTimeSeconds() - flushInterval) {
-      flush()
-    }
+  companion object {
+    const val TENANT = "tenant"
+    const val DEPTH = "tenant"
```
Comment (jerome89) [inline graphene-writer/src/main/kotlin/com/graphene/writer/store/key/SimpleKeyStoreHandler.kt]: Seems DEPTH = "tenant" has to be changed if not intended.

File: graphene-writer/src/main/kotlin/com/graphene/writer/store/key/AbstractElasticsearchKeyStoreHandler.kt [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            bulkProcessor.add(IndexRequest(index, type, metric.getTenant() + "_" + graphiteKeyPart)
+              .source(source(metric.getTenant(), graphiteKeyPart, depth, isLeaf(depth, parts))))
+          } catch (e: IOException) {
+            logger.error(e)
+          }
+        }
+      }
+    }
+
+    request = MetricMultiGetRequestBuilder(client, index, type)
```
Comment (jerome89) [inline graphene-writer/src/main/kotlin/com/graphene/writer/store/key/AbstractElasticsearchKeyStoreHandler.kt]: Do we need this line?

File: graphene-writer/src/main/kotlin/com/graphene/writer/store/key/SimpleKeyStoreHandler.kt (hunk 2 above)
Comment (Dark0096) [reply graphene-writer/src/main/kotlin/com/graphene/writer/store/key/SimpleKeyStoreHandler.kt]: Please check this commit log:
https://github.com/graphene-monitoring/graphene/pull/7/commits/0ba7b5d524fbbf02ec8ef6d68a1e826539d47ff9

File: graphene-writer/src/main/kotlin/com/graphene/writer/store/key/AbstractElasticsearchKeyStoreHandler.kt (hunk 3 above)
Comment (Dark0096) [reply graphene-writer/src/main/kotlin/com/graphene/writer/store/key/AbstractElasticsearchKeyStoreHandler.kt]: Without this logic, the request would not be initialized.
The request is to accumulate the incoming metrics and check if the key is already stored in ES. If not initialized, existing keys and new keys will continue to accumulate, causing problems such as OOM.

File: graphene-common/src/main/java/net/iponweb/disthene/reader/utils/DateTimeUtils.java (hunk 1 above)
Comment (Dark0096) [reply graphene-common/src/main/java/net/iponweb/disthene/reader/utils/DateTimeUtils.java]: The DateTimeUtils of popular library provides the currentTimeMillis() method. This means that some people are familiar with this context.
So I named the currentTimeSeconds(). Personally I think it's a good pattern to have TimeUnit included in the method name or explicitly get TimeUnit with that value like that `DateTimeUtils.currentTime(TimeUnit.Second)`

**Pull request merged**